### ISSUE TITLE
configure_make failing to configure with LuaJIT

### ISSUE THREAD

**Issue by russellsch** [bazelbuild/rules_foreign_cc#284]
I've been trying to use configure_make with LuaJIT as an external library, but keep getting stuck (possibly through my own ineptitude).

Running `make && make install` works on a local clone of LuaJIT, but in Bazel with rules_foreign_cc
fails with the following error:
> bazel-out/k8-fastbuild/bin/luajit/logs/Configure_script.sh: line 39: /home/r/.cache/bazel/_bazel_r/289d6b4dc61e219c601a4551955fed3d/sandbox/linux-sandbox/4/execroot/__main__/external/luajit/configure: No such file or directory

The lack of a configure script in LuaJIT seems to be interfering with configure_make.  I've noticed the Envoy project applies a [patch file](https://github.com/envoyproxy/envoy/blob/master/bazel/foreign_cc/luajit.patch) to get around this (which I can get to work), but I'd ideally like to find a simpler/easier way as I'm trying to encourage others in my group to use Bazel more (bringing in external deps being a main sticking point at the moment).

WORKSPACE:
```
load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
all_content = """filegroup(name = "all", srcs = glob(["**"]), visibility = ["//visibility:public"])"""
http_archive(
    name = "rules_foreign_cc",
    strip_prefix = "rules_foreign_cc-master",
    url = "https://github.com/bazelbuild/rules_foreign_cc/archive/master.zip",
)
load("@rules_foreign_cc//:workspace_definitions.bzl", "rules_foreign_cc_dependencies")
rules_foreign_cc_dependencies(register_default_tools = True)

http_archive(
    name = "luajit",
    build_file_content = all_content,
    strip_prefix = "LuaJIT-2.1.0-beta3",
    urls = ["https://github.com/LuaJIT/LuaJIT/archive/v2.1.0-beta3.tar.gz"],
)
```

BUILD:
```
load("@rules_foreign_cc//tools/build_defs:configure.bzl", "configure_make")

configure_make(
    name = "luajit",
    lib_source = "@luajit//:all",
)

```

**Comment by irengrig** [comment]
As far as I see, the code you are trying to build only has Makefile, no configure script, that is why the configure_make rule fails.

After #285, where I added "pure GNU Make" rule, **it became possible to build libraries with GNU Make with rules_foreign_cc.**
I tried to create **an example for LuaJit** you are using in the example code: https://github.com/bazelbuild/rules_foreign_cc/pull/291
Unfortunately, it does not work reliably. Probably you could have a look at the Makescript itself, I think the problem is specific to the script (and maybe my local environment).

Also, as I searched the Internet, there are LuaJit distributions there for building with CMake: https://github.com/LuaDist/luajit, maybe you could try out those with rules_foreign_cc.
However, I do not know much about this particular repository.

Hope it helps.

**Comment by lizan** [comment]
FYI in Envoy we are successfully using configure_make (with patch) to build LuaJIT.

https://github.com/envoyproxy/envoy/blob/v1.10.0/bazel/foreign_cc/BUILD#L32

**Comment by lizan** [comment]
Note some of our patch (in Envoy) is important to make LuaJIT work correctly. (esp. `-fno-function-sections -fno-data-sections`, otherwise randomly panicking when JIT occurs)

**Comment by irengrig** [comment]
Thank you for the hint!
I see - you are supplying custom build.py as the configure script, modifying/setting environment variables, for instance, TARGET_CFLAGS.

Linking to the sources in Envoy project:
a patch creating build.py: https://github.com/envoyproxy/envoy/blob/0e12efc140430033b98c9927464df2258c2aeffd/bazel/foreign_cc/luajit.patch
configure_make target for building LuaJit: https://github.com/envoyproxy/envoy/blob/ba0b5a760acc0a445df47e7cc8bdbae47312776b/bazel/foreign_cc/BUILD#L32

**Comment by irengrig** [comment]
I tried to apply some of the options, and I still observe random compilation failures. Since in my example it is not exactly the same as in Envoy's case, can be explained by those differences.
Maybe it is easier to adopt Envoy's way of building LuaJit: it works, and they also applied some patch to the Makefile.

**Comment by russellsch** [comment]
Ok, thanks you much for taking a look, I'll switch to Envoy's method

**Comment by irengrig** [comment]
Update: the problem fixed in #317 was actually causing the problems with LuaJit building.
I submitted the example in #291
